physical/oci_objectstorage/oci_objectstorage_ha.go

```diff
@@ -0,0 +1,554 @@
+		opcClientRequestId = ""
+	}
+	l.backend.logger.Debug("physical/oci: getHa", "opc-client-request-id", opcClientRequestId)
+
+	request := objectstorage.GetObjectRequest{
+		NamespaceName:      &l.backend.namespaceName,
+		BucketName:         &l.backend.lockBucketName,
+		ObjectName:         &l.key,
+		OpcClientRequestId: &opcClientRequestId,
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, ObjectStorageCallsReadTimeout)
+	defer cancel()
+
+	response, err := l.backend.client.GetObject(ctx, request)
+	l.backend.logRequest("getHA", response.RawResponse, response.OpcClientRequestId, response.OpcRequestId, err)
+
+	if err != nil {
+		if response.RawResponse != nil && response.RawResponse.StatusCode == http.StatusNotFound {
+			return nil, "", nil
+		}
+
+		metrics.SetGauge(metricGetFailed, 1)
+		l.backend.logger.Error("physical/oci: Error calling GET", "err", err)
+		return nil, "", errwrap.Wrapf(fmt.Sprintf("failed to read Value for %q: {{err}}", l.key), err)
+	}
+
+	defer response.RawResponse.Body.Close()
+
+	body, err := ioutil.ReadAll(response.Content)
+	if err != nil {
+		metrics.SetGauge(metricGetFailed, 1)
+		l.backend.logger.Error("physical/oci: Error reading content", "err", err)
+		return nil, "", errwrap.Wrapf("failed to decode Value into bytes: {{err}}", err)
+	}
+
+	var lockRecord LockRecord
+	err = json.Unmarshal(body, &lockRecord)
+	if err != nil {
+		metrics.SetGauge(metricGetFailed, 1)
+		l.backend.logger.Error("physical/oci: Error un-marshalling content", "err", err)
+		return nil, "", errwrap.Wrapf(fmt.Sprintf("failed to read Value for %q: {{err}}", l.key), err)
+	}
+
+	return &lockRecord, *response.ETag, nil
+}
+
+func (l *Lock) writeLock() (bool, error) {
+	l.backend.logger.Debug("physical/oci: WriteLock() called")
+
+	// Create a transaction to read and the update (maybe)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// The transaction will be retried, and it could sit in a queue behind, say,
+	// the delete operation. To stop the transaction, we close the context when
+	// the associated stopCh is received.
+	go func() {
+		select {
+		case <-l.stopCh:
+			cancel()
+		case <-ctx.Done():
+		}
+	}()
+
+	lockRecordCache := loadLockRecordCache(l)
+	if (lockRecordCache == nil) || lockRecordCache.lockRecord == nil || lockRecordCache.lockRecord.Identity != l.identity || time.Now().UTC().Sub(lockRecordCache.time) > l.ttl {
+		// case secondary
+		currentLockRecord, currentEtag, err := l.get(ctx)
+		if err != nil {
+			return false, errwrap.Wrapf("error reading lock record: {{err}}", err)
+		}
+
+		if (lockRecordCache == nil) || lockRecordCache.etag != currentEtag {
+			// update cached lock record
+			l.lockRecordCache.Store(&LockCache{
+				etag:       currentEtag,
+				time:       time.Now().UTC(),
+				lockRecord: currentLockRecord,
+			})
+
+			lockRecordCache = loadLockRecordCache(l)
+		}
+
+		cachedLockRecordJson, err := json.Marshal(l.lockRecordCache)
```

Comment:
The seems expensive to do all the time. I'd put these 5 lines within a logger.IsDebug conditional.

Reply:
Removed the these debug code